tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -287,6 +293,150 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
+		// create backup CRD to process backup
+		backup := fixture.GetBackupCRDWithBR(tcFrom, backupFolder)
+		_, err = cli.PingcapV1alpha1().Backups(ns).Create(backup)
+		framework.ExpectNoError(err)
+
+		// check backup is successed
+		err = wait.PollImmediate(5*time.Second, 10*time.Minute, func() (bool, error) {
+			tmpBackup, err := cli.PingcapV1alpha1().Backups(ns).Get(backup.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			// Check the status in conditions one by one,
+			// if the status other than complete or failed is running
+			for _, condition := range tmpBackup.Status.Conditions {
+				if condition.Type == v1alpha1.BackupComplete {
+					return true, nil
+				} else if condition.Type == v1alpha1.BackupFailed {
+					return false, errors.NewInternalError(nerrors.New(condition.Reason))
+				}
+			}
+			return false, nil
+		})
+		framework.ExpectNoError(err)
+
+		ginkgo.By(fmt.Sprintf("Begion to Restore data cluster %q", clusterTo.ClusterName))
+		// create restore CRD to process restore
+		restore := fixture.GetRestoreCRDWithBR(tcTo, backupFolder)
+		_, err = cli.PingcapV1alpha1().Restores(ns).Create(restore)
+		framework.ExpectNoError(err)
+
+		// check restore is successed
+		err = wait.PollImmediate(5*time.Second, 10*time.Minute, func() (bool, error) {
+			tmpRestore, err := cli.PingcapV1alpha1().Restores(ns).Get(restore.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			// Check the status in conditions one by one,
+			// if the status other than complete or failed is running
+			for _, condition := range tmpRestore.Status.Conditions {
+				if condition.Type == v1alpha1.RestoreComplete {
+					return true, nil
+				} else if condition.Type == v1alpha1.RestoreFailed {
+					return false, errors.NewInternalError(nerrors.New(condition.Reason))
+				}
+			}
+			return false, nil
+		})
+		framework.ExpectNoError(err)
+
+		ginkgo.By(fmt.Sprintf("Check the correctness of cluster %q and %q", clusterFrom.ClusterName, clusterTo.ClusterName))
+		isSame, err := oa.DataIsTheSameAs(&clusterFrom, &clusterTo)
+		framework.ExpectNoError(err)
+		if !isSame {
+			framework.ExpectNoError(nerrors.New("backup database and restore database is not the same"))
+		}
+
+		// delete backup data in S3
+		err = cli.PingcapV1alpha1().Backups(ns).Delete(backup.Name, &metav1.DeleteOptions{})
+		framework.ExpectNoError(err)
+
+		err = wait.PollImmediate(5*time.Second, 5*time.Minute, func() (bool, error) {
+			awsConfig := aws.NewConfig().
+				WithRegion(backup.Spec.S3.Region).
+				WithCredentials(cred)
+			svc := s3.New(session.Must(session.NewSession(awsConfig)))
+			input := &s3.ListObjectsV2Input{
+				Bucket: aws.String(backup.Spec.S3.Bucket),
+				Prefix: aws.String(backup.Spec.S3.Prefix),
+			}
+			result, err := svc.ListObjectsV2(input)
+			if err != nil {
+				return false, err
+			}
+			if *result.KeyCount != 0 {
+				return false, nil
+			}
+			return true, nil
+		})
```

Comment:
because we share this S3 bucket, we can't check all files in this bucket because it's possible another job is writing files into it.
can you configure a prefix? you can generate a random prefix at the beginning and use it in each test.

Reply:
actually the prefix is defined in the begining of the test as backupFolder, and the listObjectv2 is check about files with this prefix

Reply:
Great! I missed it.